easybuild/easyblocks/g/gromacs.py

```diff
@@ -493,3 +563,118 @@ def sanity_check_step(self):
+                'double': '--enable-double',
+            }
+            mpi_type_opts = {
+                'nompi': '--disable-mpi',
+                'mpi': '--enable-mpi'
+            }
+        else:
+            prec_opts = {
+                'single': '-DGMX_DOUBLE=OFF',
+                'double': '-DGMX_DOUBLE=ON',
+            }
+            mpi_type_opts = {
+                'nompi': '-DGMX_MPI=OFF -DGMX_THREAD_MPI=ON',
+                'mpi': '-DGMX_MPI=ON -DGMX_THREAD_MPI=OFF'
+            }
+
+        # Double precision pattern so search for in configopts
+        self.DP_pattern = prec_opts['double']
+
+        if LooseVersion(self.version) < LooseVersion('5'):
+            # For older versions we only build/install the mdrun part for
+            # the MPI variant. So we need to be able to specify the
+            # install target depending on variant.
+            self.cfg['install_cmd'] = 'make'
+
+            # Use the fact that for older versions we just need to
+            # build and install mdrun for the MPI part
+            build_opts = {
+                'nompi': '',
+                'mpi': 'mdrun'
+            }
+            install_opts = {
+                'nompi': 'install',
+                'mpi': 'install-mdrun'
+            }
+        else:
+            build_opts = {
+                'nompi': '',
+                'mpi': ''
+            }
+            install_opts = {
+                'nompi': 'install',
+                'mpi': 'install'
+            }
+
+        precisions = ['single']
+        if self.cfg.get('double_precision') is None or self.cfg.get('double_precision'):
+            precisions.append('double')
+
+        mpitypes = ['nompi']
+        if self.toolchain.options.get('usempi', None):
+            mpitypes.append('mpi')
+
+        # We need to count the numb er of variations to build.
+        self.variants_to_build = 0
+        versions_built = []
+        # Handle the different variants
+        for precicion in precisions:
+            for mpitype in mpitypes:
+                versions_built.append('%s precision %s' % (precicion, mpitype))
+                var_confopts = []
+                var_buildopts = []
+                var_installopts = []
+
+                var_confopts.append(mpi_type_opts[mpitype])
+                var_confopts.append(prec_opts[precicion])
+                if LooseVersion(self.version) < LooseVersion('4.6'):
+                    suffix = ''
+                    if mpitype == 'mpi':
+                        suffix = "--program-suffix={0}".format(self.cfg.get('mpisuffix'))
+                        if precicion == 'double':
+                            suffix += '_d'
+                    var_confopts.append(suffix)
+
+                var_buildopts.append(build_opts[mpitype])
+                var_installopts.append(install_opts[mpitype])
+
+                self.cfg.update('configopts', [' '.join(var_confopts) + ' ' + common_config_opts])
```

Comment:
I think there's no need for the `[...]`? And you can avoid the `+ ' ' +` too:

```python
self.cfg.update('configopts', ' '.join(var_confopts + [common_config_opts]))

Reply:
configopts is a list here, don't I need to give it a list then?

Reply:
No, it will leverage it to a list automatically, see https://github.com/easybuilders/easybuild-framework/blob/develop/easybuild/framework/easyconfig/easyconfig.py#L575

Reply:
Done